chasm/lib/scheduler/scheduler.go

```diff
@@ -582,3 +604,63 @@ func (s *Scheduler) validateConflictToken(token []byte) bool {
 	current := s.generateConflictToken()
 	return bytes.Equal(current, token)
 }
+
+// SearchAttributes returns the Temporal-managed key values for visibility.
+func (s *Scheduler) SearchAttributes(chasm.Context) []chasm.SearchAttributeKeyValue {
+	return []chasm.SearchAttributeKeyValue{
+		chasm.SearchAttributeTemporalSchedulePaused.Value(s.Schedule.GetState().GetPaused()),
+	}
+}
+
+// Memo returns the scheduler's info block for visibility.
+func (s *Scheduler) Memo(
+	ctx chasm.Context,
+) map[string]chasm.VisibilityValue {
+	newInfo, err := s.ListInfo(ctx)
+	if err != nil {
+		// Unable to retrieve list info. Return nil to skip memo update.
+		// Error will be logged once loggers are available in CHASM.
+		return nil
+	}
+
+	payload, err := newInfo.Marshal()
+	if err != nil {
+		// Unable to marshal list info. Return nil to skip memo update.
+		// Error will be logged once loggers are available in CHASM.
+		return nil
+	}
```

Comment:
Unrelated to this PR but both places where this could error out previously will not be relevant once the the next visibility change is checked in making the logger comment irrelevant.